commitlint/plugins.ts

```diff
@@ -399,7 +399,20 @@ export abstract class Plugins {
 
                         let lineIsFooterNote = Helpers.isFooterNote(line);
 
-                        if (!isUrl && !lineIsFooterNote) {
+                        let endsWithCommitHashButRestIsNotTooLong = false;
+                        let commitHashAtTheEndMatch =
+                            line.match(`([0-9a-f]{40})\.?$`);
```

Comment:
@webwarrior-ws let's make this pattern match not only when a dot is at the end, but also any punctuation sign (comma, colon, semicolon) and even no character.

Reply:
(also exclamation and interrogation: ! ?)

Reply:
Included other punctuation signs in the regex.

Reply:
Oops, I've accidentally overwritten my changes to regex when replacing match() with search(). Now it is fixed.